### PR TITLE
Added new tests

### PR DESCRIPTION
**Pull request by iotcloudarchitect**
# New Pull Request

For more information, see the `CONTRIBUTING` guide.

## Prerequisites

Please make sure you can check the following boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [x] All new and existing tests passed

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [ ] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have updated the documentation accordingly
- [x] I have added tests to cover my changes

### Description

Tests for components/Mediator and utils/foreign

### Remarks

[Optionally place any follow-up comments, remarks, observations, or notes here for future reference]

### CODE REVIEW

File: src/framework/piral-core/src/utils/foreign.test.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { createElement, createRef } from 'react';
+import { convertComponent, renderInDom } from './foreign';
+import { ForeignComponent } from '../types';
+import { DefaultLoadingIndicator } from '../components/DefaultLoader';
+
+describe('Util Foreign.', () => {
+  it('Convert component function throws error due to missing converter function.', () => {
+    try {
```
Comment (FlorianRappl) [inline src/framework/piral-core/src/utils/foreign.test.ts:8]: Hm this is a little bit shacky here. Normally, you'd supply a function here:

```ts
test("Test description", () => {
  const t = () => {
    throw new TypeError();
  };
  expect(t).toThrow(TypeError);
});
```

File: src/framework/piral-core/src/utils/foreign.test.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      expect(convertComponent(null, createElement('div'))).toThrow(
+        'No converter for component of type "div" registered.',
+      );
+    } catch {
+      expect(true).toEqual(true);
+    }
+  });
+  it('Convert component function returns converted component.', () => {
```
Comment (FlorianRappl) [inline src/framework/piral-core/src/utils/foreign.test.ts:16]: Would appreciate a blank line between the tests - just to be able to see immediately what's going on.

Thanks!

Comment (iotcloudarchitect) [reply src/framework/piral-core/src/utils/foreign.test.ts:14]: Done. Thanks for hint.

File: src/framework/piral-core/src/utils/foreign.test.ts (hunk 1 above)
Comment (iotcloudarchitect) [reply src/framework/piral-core/src/utils/foreign.test.ts]: Hope it suits now. Thanks for the hint.

Comment (FlorianRappl) [reply src/framework/piral-core/src/utils/foreign.test.ts]: Yes, great job! :+1:

**Pull request merged**